### DISCUSSION TITLE
Parcel compilation with build command

### DISCUSSION THREAD

**Discussion by ghost** [parcel-bundler/parcel#4652] [General]
### Hello everybody, I'd like to know two things: 

First one is if this kind of compilation is usual in Parcel, when I try the compiled app.js in the browser console it tells me that Parcel is not defined and the mined file doesn't do it well.

>PARCEL BUILD
![parcelBuild](https://user-images.githubusercontent.com/22520811/82756131-79ca0180-9dd8-11ea-8f02-9e0de437a3b2.jpg)

**I don't understand why Parcel leaves that gap in the file obfuscated and mined.**

> WEBPACK BUILD
![webpackBuild](https://user-images.githubusercontent.com/22520811/82756132-7afb2e80-9dd8-11ea-9a41-5209f41fabdc.jpg)

The second I wanted to know is if the tree shaking is activated by default, I attach a comparison of a compilation made by Webpack and the one made by Parcel.

**Comment by mischnic** [comment]
> it tells me that Parcel is not defined and the mined file doesn't do it well.

Which error are you getting? `parcelRequire is not defined`? The output you posted doesn't work in `<script type="module">`

> leaves that gap in the file obfuscated and mined.

Looks like the newline there is just hardcoded:

https://github.com/parcel-bundler/parcel/blob/d33fcd9cf826188288e7ffb5dfecbfa221592458/packages/core/parcel-bundler/src/packagers/JSPackager.js#L113-L121

(And Parcel 1 ran Terser per file, not on the whole bundle. So Terser can't remove it either)

> The second I wanted to know is if the tree shaking is activated by default

Parcel 1: no, add `--experimental-tree-shaking`
Parcel 2: yes

**Reply by ghost** [reply]
Yes, that error I'm getting but the minied script is wrong with a nbsp
